src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -157,11 +157,18 @@ object SchemaUtils {
     }
 
     fun drop(vararg tables: Table) {
-        EntityCache.sortTablesByReferences(tables.toList()).reversed()
-                .filter { it in tables && it.exists() }
+        var tablesForDeletion = EntityCache
+                .sortTablesByReferences(tables.toList())
+                .reversed()
+                .filter { it in tables }
+        val transactionManager = TransactionManager.current()
+        if (!transactionManager.db.dialect.supportsIfNotExists) {
```

Comment:
There is `currentDialect` shortcut for this.

Reply:
Fixed.